src/zodbverify/verify_oid.py

```diff
@@ -13,6 +20,173 @@
 logger = logging.getLogger("zodbverify")
 
 
+class Refbuilder(object):
+
+    def __init__(self, storage, connection):
+        self.storage = storage
+        self.connection = connection
+        self.seen = []
+        self.refs = defaultdict(list)
+        self.msg = []
+        self.stop_recurse = [
+            '0x00',  # persistent.mapping.PersistentMapping
+            '0x01',  # OFS.Application.Application
+            '0x11',  # Products.CMFPlone.Portal.PloneSite
+        ]
+
+    def get_oid_report(self, oid, level=0, max_level=600, verbose=False):
+        self.msg.append('The oid {} is referenced by:\n'.format(oid))
+        self.inspect_reference_tree(oid, level=0, max_level=600, verbose=verbose)
+        logger.info('\n'.join(self.msg))
+
+    def _build_ref_tree(self):
+        logger.info('Building a reference-tree of ZODB...')
+        count = 0
+        next_ = None
+        while True:
+            count += 1
+            oid, tid, data, next_ = self.storage.record_iternext(next_)
+
+            # For each oid create a list of oids that reference it
+            # Can be used for reverse lookup similar to what fsoids.py
+            # in ZODB does.
+            oid_refs = get_refs(data)
+            if oid_refs:
+                for referenced_oid, class_info in oid_refs:
+                    self.refs[oid_repr(referenced_oid)].append(oid_repr(oid))
+            if next_ is None:
+                break
+            if not count % 10000:
+                logger.info('Objects: {}'.format(count))
+        logger.info('Created a reference-dict for {} objects.\n'.format(count))
+
+    def inspect_reference_tree(self, oid, level=0, max_level=600, verbose=False):
+        if oid not in self.refs:
+            logger.debug('The oid {} does not exist!'.format(oid))
+            return
+        child_pickle, state = self.storage.load(repr_to_oid(oid))
+        child_class_info = '%s.%s' % get_pickle_metadata(child_pickle)
+
+        for ref in self.refs[oid]:
+            if ref in self.seen:
+                continue
+            level += 1
+            if level > max_level:
+                msg = '8< --------------- >8 Stop after level {}!\n'.format(max_level)
+                self.msg.append(msg)
+                logger.debug(msg)
+                continue
+            self.seen.append(ref)
+            pick, state = self.storage.load(repr_to_oid(ref))
+            class_info = '%s.%s' % get_pickle_metadata(pick)
+            name = None
+            if verbose:
+                name = self.get_id_or_attr_name(oid=oid, parent_oid=ref)
+
+            if name:
+                msg = '{} ({}) is {} for {} ({}) at level {}'.format(oid, child_class_info, name, ref, class_info, level)
```

Comment:
With that many parameters kwargs are a better way to pass parameters to `.format`.